test/e2e/cmd/workspaces_test.go

```diff
@@ -36,57 +32,96 @@ import (
 const (
 	testResultsDirectory = "/tmp/artifacts"
 	jUnitOutputFilename  = "junit-workspaces-operator.xml"
+	testServiceAccount   = "terminal-test"
 )
 
 //SynchronizedBeforeSuite blocks is executed before run all test suites
 var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
-	fmt.Println("Starting to setup objects before run ginkgo suite")
-	config.Namespace = "devworkspace-controller"
+	log.Println("Starting to setup objects before run ginkgo suite")
 
-	k8sClient, err := client.NewK8sClient()
-	if err != nil {
-		fmt.Println("Failed to create workspace client")
-		panic(err)
+	var err error
+	kubeConfig := os.Getenv("KUBECONFIG")
+
+	if len(kubeConfig) == 0 {
+		failMess := "The mandatory environment variable(s) is not set.\nMake sure that all variables have been set properly. " +
+			"The variable list:\nKUBECONFIG=%s"
+		ginkgo.Fail(fmt.Sprintf(failMess, kubeConfig))
 	}
 
-	controller := deploy.NewDeployment(k8sClient)
+	config.AdminK8sClient, err = client.NewK8sClientWithKubeConfig(kubeConfig)
 
-	err = controller.CreateNamespace()
 	if err != nil {
-		panic(err)
+		ginkgo.Fail("Cannot create admin k8s client. Cause: " + err.Error())
 	}
 
-	if err := controller.DeployWorkspacesController(); err != nil {
-		fmt.Println("Failed to deploy workspace controller")
-		panic(err)
-	}
+	config.OperatorNamespace = "devworkspace-controller"
+	config.WorkspaceNamespace = "test-terminal-namespace"
 
-	return nil
-}, func(data []byte) {})
+	//create the test workspace for the test user under kube admin
 
-var _ = ginkgo.SynchronizedAfterSuite(func() {
-	k8sClient, err := client.NewK8sClient()
+	err = config.AdminK8sClient.CreateNamespace(config.WorkspaceNamespace)
+	if err != nil {
+		ginkgo.Fail(fmt.Sprintf("Cannot create the namespace %q: Cause: %s", config.WorkspaceNamespace, err.Error()))
+	}
+
+	err = config.AdminK8sClient.CreateSA(testServiceAccount, config.WorkspaceNamespace)
+	if err != nil {
+		ginkgo.Fail("Cannot create test SA. Cause: " + err.Error())
 
+	}
+	err = config.AdminK8sClient.AssignRoleToSA(config.WorkspaceNamespace, testServiceAccount, "admin")
 	if err != nil {
-		_ = fmt.Errorf("Failed to uninstall workspace controller %s", err)
+		ginkgo.Fail("Cannot create test rolebinding for SA. Cause: " + err.Error())
 	}
 
-	if err = k8sClient.Kube().AdmissionregistrationV1().MutatingWebhookConfigurations().Delete(context.TODO(), workspaceWebhook.MutateWebhookCfgName, metav1.DeleteOptions{}); err != nil {
-		_ = fmt.Errorf("Failed to delete mutating webhook configuration %s", err)
+	token, err := config.AdminK8sClient.GetSAToken(config.WorkspaceNamespace, testServiceAccount)
+	if token == "" {
+		//sometimes the Service Account token is not applied immediately in this case we wait 1 sec.
+		// and try obtain it again
+		time.Sleep(1 * time.Second)
```

Comment:
I wonder if it's worth it to wait a few extra seconds here just so that line 87 shouldn't fail (I don't know how often it fails though)

Reply:
Maybe then it's better to rework it with a couple of tries as we have with waiting devws status.